### DISCUSSION TITLE
Unable to get Model Binding to work

### DISCUSSION THREAD

**Discussion by abishekrsrikaanth** [livewire/livewire#3755] [1. Help] [answered]
I have the following code

```php
class BusinessProfile extends Component
{
    public Supplier $supplier;
    public $name;

    protected $rules = [];

    public function mount()
    {
        $user = auth()->user();

        $this->supplier = $user->supplier;

        $this->name = $this->supplier->business_name;
    }

    public function render()
    {
        return view('livewire.settings.business-profile');
    }
}
```

and the view
```html
   <input type="text" wire:model="supplier.business_name">
   <input type="text" wire:model="name">
```

The input field with the model binding is empty but the `public` variable binding has the right value. What is causing this?
Is there anything specific that I need to do to enable model binding in Livewire?

**Comment by joshhanley** [comment, marked as answer]
@abishekrsrikaanth you need to make sure you have rules if you are binding to eloquent models 🙂 see here https://laravel-livewire.com/docs/2.x/properties#binding-models

> Note: For this to work, you have a validation entry in the `$rules` property for any model attributes you want to bind to. Otherwise, an error will be thrown.

Hope this helps!
